<launch>
	<arg name="marker_size" default="6.22" />
	<arg name="max_new_marker_error" default="0.08" />
	<arg name="max_track_error" default="0.2" />
	<arg name="cam_image_topic" default="/myrobot/body_camera/color/image_raw" />
	<arg name="cam_info_topic" default="/myrobot/body_camera/color/camera_info" />
	<arg name="output_frame" default="/body_camera_color_optical_frame" />
  <arg name="bundle_files" default="$(find detect)/config/markers.xml" />

	<node name="ar_track_alvar" pkg="ar_track_alvar" type="findMarkerBundlesNoKinect" respawn="false" output="screen" 
        args="$(arg marker_size) 
              $(arg max_new_marker_error) 
              $(arg max_track_error) 
              $(arg cam_image_topic) 
              $(arg cam_info_topic) 
              $(arg output_frame) 
              $(arg bundle_files)" />

<!-- 
	<node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg output_frame)" />

		<remap from="camera_image"  to="$(arg cam_image_topic)" />
		<remap from="camera_info"   to="$(arg cam_info_topic)" />
	</node>
 -->


	<node name="container_tf_adjuster" pkg="detect" type="container_tf_adjuster.py" />

  <node pkg="tf" type="static_transform_publisher" name="container_br" args="0 -0.59 0.12 0 0 0 /container_origin /container_br 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_tr" args="1 -0.59 0.12 0 0 0 /container_origin /container_tr 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_tl" args="1 0.59 0.12 0 0 0 /container_origin /container_tl 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_bl" args="0 0.59 0.12 0 0 0 /container_origin /container_bl 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_base" args="0.5 0 -0.78 0 0 0 /container_origin /container_base 30"/>

<!-- 

  <node pkg="tf" type="static_transform_publisher" name="container_br" args="0.57 0.13 0 0 0 0 /container_origin /container_br 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_tr" args="0.57 0.13 -1.0 0 0 0 /container_origin /container_tr 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_bl" args="-0.61 0.13 0 0 0 0 /container_origin /container_bl 30"/>
  <node pkg="tf" type="static_transform_publisher" name="container_tl" args="-0.61 0.13 -1.0 0 0 0 /container_origin /container_tl 30"/>

 -->


</launch>
